drivers/python/rethinkdb/_export.py

```diff
@@ -188,12 +205,21 @@ def write_table_metadata(progress, conn, db, table, base_path):
 # This is called through rdb_call_wrapper and may be called multiple times if
 # connection errors occur.  In order to facilitate this, we do an order_by by the
 # primary key so that we only ever get a given row once.
-def read_table_into_queue(progress, conn, db, table, pkey, task_queue, progress_info, exit_event):
+def read_table_into_queue(progress, conn, db, table, pkey, keys, index, task_queue, progress_info, exit_event):
     read_rows = 0
-    if progress[0] is None:
-        cursor = r.db(db).table(table).order_by(index=pkey).run(conn, time_format="raw", binary_format='raw')
+    table = r.db(db).table(table)
+
+    if progress[0] is not None:
+        table = table.between(progress[0], None, left_bound="open")
+
+    if keys is None:
+        table = table.order_by(index=pkey)
+    elif index is not None:
+        table = table.get_all(*keys, index=index).order_by(pkey)
     else:
-        cursor = r.db(db).table(table).between(progress[0], None, left_bound="open").order_by(index=pkey).run(conn, time_format="raw", binary_format='raw')
+        table = table.get_all(*keys).order_by(pkey)
```

Comment:
I tried that, but it turns out that RethinkDB actually doesn't support indexed `order_by` on `get_all` queries (or indeed any non-table selection).

Doing the `get_all` after the `order_by` doesn't work either, because `get_all` is defined only on tables.